drivers/scheduler/eks/eks.go

```diff
@@ -0,0 +1,304 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/eks/types"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/sched-ops/task"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	kube "github.com/portworx/torpedo/drivers/scheduler/k8s"
+	"github.com/portworx/torpedo/pkg/log"
+	"os"
+	"strings"
+	"time"
+)
+
+const (
+	// SchedName is the name of the kubernetes scheduler driver implementation
+	SchedName = "eks"
+	// defaultEKSUpgradeTimeout is the default timeout for EKS control plane upgrade
+	defaultEKSUpgradeTimeout = 90 * time.Minute
+	// defaultEKSUpgradeRetryInterval is the default retry interval for EKS control plane upgrade
+	defaultEKSUpgradeRetryInterval = 5 * time.Minute
+)
+
+type EKS struct {
+	kube.K8s
+	clusterName     string
+	region          string
+	config          aws.Config
+	eksClient       *eks.Client
+	pxNodeGroupName string
+}
+
+// String returns the string name of this driver.
+func (e *EKS) String() string {
+	return SchedName
+}
+
+func (e *EKS) Init(schedOpts scheduler.InitOptions) (err error) {
+	// This implementation assumes the EKS cluster has two node groups: one group for
+	// Torpedo and another group for Portworx.
+	err = e.K8s.Init(schedOpts)
+	if err != nil {
+		return err
+	}
+	torpedoNodeName := ""
+	pods, err := core.Instance().GetPods("default", nil)
+	if err != nil {
+		log.Errorf("failed to get pods from default namespace. Err: [%v]", err)
+	}
+	if pods != nil {
+		for _, pod := range pods.Items {
+			if pod.Name == "torpedo" {
+				torpedoNodeName = pod.Spec.NodeName
+			}
+		}
+	}
+	nodes, err := core.Instance().GetNodes()
+	if err != nil {
+		log.Errorf("failed to get nodes. Err: [%v]", err)
+	}
+	e.region = os.Getenv("EKS_CLUSTER_REGION")
```

Comment:
Do not add this as part of `init()`, move it to `UpgradeScheduler()`, as this is the place where its needed

Reply:
If I add more methods to the EKS Scheduler in the future, wouldn't it be repetitive to use the same code again in the new function? I could create a separate function to avoid this, but even then, repeatedly calling it might feel unnecessary.

Reply:
Got it. Done. Thanks.